### PR TITLE
Improve SQL editor and chat-panel integration

### PR DESCRIPTION
**Pull request by FrankChen021**
## Summary
- add floating SQL-selection actions in the editor for run, explain, add to chat, and AI commands
- add inline SQL snippet tokens in the chat composer with hover preview and correct fenced-markdown expansion on send
- simplify query control agent actions into a toggle button and fix token hover/removal edge cases

## Validation
- npm run format
- npm run typecheck
- npm run lint
- npm run test -- src/components/chat/input/inline-token.test.ts src/components/chat/input/sql-snippet-token.test.ts src/components/chat/message/message-user.test.tsx

## Notes
- npm run build is blocked in this worktree because vendored directories are missing here: external/number-flow/packages/number-flow, external/clickhouse/skills, and external/vizlayer/skills

### CODE REVIEW

File: src/components/chat/message/message-user.tsx [hunk 1]
```diff
@@ -36,7 +36,10 @@ export const MessageUser = memo(function MessageUser({ text }: { text: string })
         if (codeBlock) return codeBlock;
         // If it's a newline outside code block, double it
         return "\n\n";
-      });
+      })
+      // Fenced code blocks must start after a blank line, otherwise markdown may render
+      // the opening backticks inline when prose appears immediately before the fence.
+      .replace(/([^\n])\n(```)/g, "$1\n\n$2");
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply src/components/chat/message/message-user.tsx]: Fixed in `72dbcf3e`. I changed the user-message preprocessing so prose normalization now runs only on non-fenced segments, which avoids mutating content inside fenced code blocks. I also updated the test to assert the fence contents stay unchanged.

File: src/components/chat/view/use-chat-panel.tsx [hunk 2]
```diff
@@ -164,8 +170,12 @@ export function ChatPanelProvider({ children }: { children: React.ReactNode }) {
...
   };
 
-  const setInitialInput = (text: string, chatId?: string) => {
-    setInitialInputState({ text, chatId });
+  const setInitialInput = (
+    text: string,
+    chatId?: string,
+    mode: ChatComposerInputMode = "replace"
+  ) => {
+    setInitialInputState({ text, chatId, mode, nonce: Date.now() });
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply src/components/chat/view/use-chat-panel.tsx]: Fixed in `72dbcf3e`. `setInitialInput()` now uses a monotonic `useRef` counter for `nonce` instead of `Date.now()`, so repeated composer updates cannot collide within the same millisecond. I also added focused coverage in `use-chat-panel.test.tsx` with `Date.now()` pinned to a constant.

File: src/components/chat/view/chat-view.tsx [hunk 3]
```diff
@@ -149,7 +150,7 @@ export const ChatView = forwardRef<ChatViewHandle, ChatViewProps>(function ChatV
         handleSubmit({ text: input.text });
         return;
       }
-      setPromptInput(input.text);
+      setPromptInput({ text: input.text, mode: "replace", nonce: Date.now() });
     },
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply src/components/chat/view/chat-view.tsx]: Fixed in `72dbcf3e`. `ChatView` now uses its own monotonic counter when creating local `promptInput` updates instead of relying on `Date.now()`, so rapid repeated sample-question/user-action updates remain distinct.

File: src/components/query-tab/query-input/query-input-view.tsx [hunk 4]
```diff
@@ -57,6 +80,18 @@ interface QueryInputViewProps {
   onRun?: (sql: string) => void;
 }
 
+type SelectionActionState = {
+  sql: string;
+  top: number;
+  left: number;
+  placement: "above" | "below";
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply src/components/query-tab/query-input/query-input-view.tsx]: Cleaned up in `72dbcf3e`. I removed the unused `placement` field from `SelectionActionState` and the corresponding write path so the selection-action state only carries values that actually drive rendering.

File: src/components/chat/input/command-utils.ts [hunk 5]
```diff
@@ -20,10 +20,17 @@ export function getLeadingCommand(text: string): LeadingCommandMatch | null {
...
+  input: string,
+  commandName: string,
+  cursorOffset: number = input.length
+): string {
+  const safeCursorOffset = Math.max(0, Math.min(cursorOffset, input.length));
+  const beforeCursor = input.slice(0, safeCursorOffset);
+  const afterCursor = input.slice(safeCursorOffset);
+  const match = LEADING_COMMAND_PREFIX_RE.exec(beforeCursor);
+  const argsStart = match ? match[0].length : beforeCursor.length;
+  const existingArgs = `${beforeCursor.slice(argsStart)}${afterCursor}`;
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply src/components/chat/input/command-utils.ts]: Fixed in `184c8ff6`. `replaceLeadingCommand()` now drops the entire current leading command token before preserving the remaining suffix/arguments, so selecting a command while the cursor is inside `/revw...` produces `/review ...` instead of `/revieww ...`. I also added a regression test for the cursor-inside-command case.
